aeneas/src/mach/GlobalRegAlloc.v3

```diff
@@ -0,0 +1,603 @@
+// Copyright 2023 Virgil Authors. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+def verbose = false;
+def colorVerbose = false;
+// Implement global register allocation with graph coloring. Build an interference
+// graph from liveness analysis.
+class GlobalRegAlloc(regSet: MachRegSet, codegen: SsaMachGen) {
+	def regState = Array<int>.new(regSet.physRegs);         // indices into {allocated}
+	def allocated = Array<(VReg, int)>.new(regSet.physRegs);    // (VReg, pos) live register
+	def regUsage = Array<int>.new(regSet.physRegs);
+	def moves = GlobalMoveSet.new(regSet);
+	def movesPrev = GlobalMoveSet.new(regSet);
+	def parmoveDests = Vector<(VReg, List<Operand.Def>)>.new();
+	def backupAllocate = LocalRegAlloc.new(regSet, codegen).allocate;
+	def availableColor = regSet.regSets[regSet.regClasses[RegClass.REF.tag]].length - 2;
+	def vars = codegen.vars;
+	var liveness: BitMatrix;
+	var buf = ArchInstrBuffer.new(codegen, codegen.mach.prog, regSet);
+	var liveout: BitMatrix;
+	var cursor: int;
+	var pos: int;
+	var prev: ArchInstr;
+	var cur: ArchInstr;
+	var next: ArchInstr;
+	var graph: Graph;
+	var visitedCnt = 0;
+	var coloring: Array<byte>;
+	var blockWeight: Array<int>;
+	var liveoutRow: int;
+
+	def clear() {
+		visitedCnt = 0;
+		for (i < allocated.length) allocated[i] = (null, -1);
+		for (i < regState.length) regState[i] = -1;
+		for (i < regUsage.length) regUsage[i] = 0;
+		cursor = 0;
+		pos = 0;
+	}
+	def allocate() {
```

Comment:
Let's put this method first and move `clear()` down. Also, add a sentence to `allocate()` to explain the overall algorithm.